android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayoutImpl.java

```diff
@@ -605,6 +632,120 @@ public void didSelectTab(Tab tab, @TabSelectionType int type, int lastId) {
+            BraveActivity braveActivity = BraveActivity.getBraveActivity();
+            ViewGroup viewGroup =
+                    braveActivity.getWindow().getDecorView().findViewById(android.R.id.content);
+
+            PlaylistOptionsListener playlistOptionsListener = new PlaylistOptionsListener() {
+                @Override
+                public void onOptionClicked(PlaylistOptionsModel playlistOptionsModel) {
+                    if (playlistOptionsModel.getOptionType() == PlaylistOptions.ADD_MEDIA) {
+                        org.chromium.url.mojom.Url contentUrl = new org.chromium.url.mojom.Url();
+                        contentUrl.url = url;
+                        int mediaCount = SharedPreferencesManager.getInstance().readInt(
+                                PlaylistPreferenceUtils.ADD_MEDIA_COUNT);
+                        if (mediaCount <= 2) {
+                            SharedPreferencesManager.getInstance().writeInt(
+                                    PlaylistPreferenceUtils.ADD_MEDIA_COUNT, mediaCount + 1);
+                        }
+                        if (mediaCount == 2) {
+                            if (BraveActivity.getBraveActivity() != null) {
+                                PlaylistWarningDialogListener playlistWarningDialogListener =
+                                        new PlaylistWarningDialogListener() {
+                                            @Override
+                                            public void onActionClicked() {
+                                                addMediaToPlaylist(contentUrl, viewGroup);
+                                            }
+
+                                            @Override
+                                            public void onSettingsClicked() {
+                                                braveActivity.openBravePlaylistSettings();
+                                            }
+                                        };
+                                BraveActivity.getBraveActivity().showPlaylistWarningDialog(
+                                        playlistWarningDialogListener);
+                            }
+                        } else {
+                            addMediaToPlaylist(contentUrl, viewGroup);
+                        }
+                    } else if (playlistOptionsModel.getOptionType()
+                            == PlaylistOptions.OPEN_PLAYLIST) {
+                        try {
+                            BraveActivity.getBraveActivity().openPlaylistActivity(
+                                    getContext(), ConstantUtils.DEFAULT_PLAYLIST);
+                        } catch (ActivityNotFoundException e) {
+                            Log.e(TAG, "Open playlist " + e);
+                        }
+                    } else if (playlistOptionsModel.getOptionType()
+                            == PlaylistOptions.PLAYLIST_SETTINGS) {
+                        braveActivity.openBravePlaylistSettings();
```

Comment:
it's better to ` BraveActivity.getBraveActivity()` as `braveActivity` outside of click scope and could be destroyed already.